SmartDeviceLink/SDLCreateWindow.h

```diff
@@ -0,0 +1,85 @@
+ @param associatedServiceType Allows an app to create a widget related to a specific service type. As an example if a `MEDIA` app becomes active, this app becomes audible and is allowed to play audio. Actions such as skip or play/pause will be directed to this active media app. In case of widgets, the system can provide a single "media" widget which will act as a placeholder for the active media app. It is only allowed to have one window per service type. This means that a media app can only have a single MEDIA widget. Still the app can create widgets omitting this parameter. Those widgets would be available as app specific widgets that are permanently included in the HMI. This parameter is related to widgets only. The default main window, which is pre-created during app registration, will be created based on the HMI types specified in the app registration request.
+ */
+- (instancetype)initWithId:(UInt32)windowId windowName:(NSString *)windowName windowType:(SDLWindowType)windowType associatedServiceType:(nullable NSString *)associatedServiceType;
+
+
+/**
+ Convinience constructor with all the parameters.
+
+ @param windowId The type of the window to be created. Main window or widget.
+ @param windowName The window name to be used by the HMI. The name of the pre-created default window will match the app name. Multiple apps can share the same window name except for the default main window. Creating a window with a name which is already in use by the app will result in `DUPLICATE_NAME`.
+ MaxLength 100.
+ @param windowType The type of the window to be created. Main window or widget.
+ @param associatedServiceType Allows an app to create a widget related to a specific service type. As an example if a `MEDIA` app becomes active, this app becomes audible and is allowed to play audio. Actions such as skip or play/pause will be directed to this active media app. In case of widgets, the system can provide a single "media" widget which will act as a placeholder for the active media app. It is only allowed to have one window per service type. This means that a media app can only have a single MEDIA widget. Still the app can create widgets omitting this parameter. Those widgets would be available as app specific widgets that are permanently included in the HMI. This parameter is related to widgets only. The default main window, which is pre-created during app registration, will be created based on the HMI types specified in the app registration request.
+ @param duplicateUpdatesFromWindowID  Optional parameter. Specify whether the content sent to an existing window should be duplicated to the created window. If there isn't a window with the ID, the request will be rejected with `INVALID_DATA`.
+ */
+- (instancetype)initWithId:(NSInteger)windowId windowName:(NSString *)windowName windowType:(SDLWindowType)windowType associatedServiceType:(nullable NSString *)associatedServiceType duplicateUpdatesFromWindowID:(UInt32)duplicateUpdatesFromWindowID;
```

Comment:
Thanks @joeljfischer it was my mistake it should be UInt32. Sorry for this.

Reply:
I think this is due to `SDLIntEnum` (based on `NSInteger` because int enums are usually NSInteger). The idea is to avoid compiler warnings between windowId and `SDLPredefinedWindows`.  I think if we create `SDLUInt32Enum` and use it for the enum we align the type.

Do you see a benefit using UInt32 instead of NSInteger? It's is closer to the machine arch and used very often by iOS frameworks and enums.

Reply:
`NSUInteger` is fine to me. Your suggested enum also looks good to me (`NS_TYPED_ENUM` seems more close to the other enums but it's rather an overkill for predefined windows enum...).
1. We'll remove SDLEnumTypes.h and revert SDLEnum.h
2. SDLPredefinedWindows.h should include your enum suggestion
3. SDLPredefinedWindows.m should be removed (not needed anymore)
4. windowID parameters in init should be of type `NSUInteger`
5. windowID properties should be of type `NSNumber<SDLUInt> *`